client/payment-request/index.js

```diff
@@ -429,6 +438,14 @@ jQuery( ( $ ) => {
 				wcpayPaymentRequest.addToCart();
 			} );
 
+			// WooCommerce Deposits support.
+			// Trigger the "woocommerce_variation_has_changed" event when the deposit option is changed.
+			$( 'input[name=wc_deposit_option]' ).on( 'change', () => {
+				$( 'form:has(input[name=wc_deposit_option])' ).trigger(
```

Comment:
```suggestion
				$( 'form' ).has('input[name=wc_deposit_option],input[name=wc_deposit_payment_plan]').trigger(
```

As above. 

I've switched to using the `has()` jQuery method rather than the selector as I don't think some browsers support selector lists in early `:has()` implementations. But I may be hallucinating.